src/DotNet/Importer.cs

```diff
@@ -374,7 +380,10 @@ public Importer(ModuleDef module, ImporterOptions options, GenericParamContext g
 			var pkt = asmName.GetPublicKeyToken();
 			if (pkt is null || pkt.Length == 0)
 				pkt = null;
-			return module.UpdateRowId(new AssemblyRefUser(asmName.Name, asmName.Version, PublicKeyBase.CreatePublicKeyToken(pkt), asmName.CultureInfo.Name));
+			if (TryToUseExistingAssemblyRefs)
```

Comment:
The part after `??` is a duplicate of the following `else`. You can try `if (TryToUseExistingAssemblyRefs && module.GetAssemblyRef(asmName.Name) is AssemblyRef asmRef) ...`